src/main/java/org/jenkinsci/maven/plugins/hpi/InitializeMojo.java

```diff
@@ -0,0 +1,75 @@
+package org.jenkinsci.maven.plugins.hpi;
+
+import hudson.util.VersionNumber;
+import io.jenkins.lib.versionnumber.JavaSpecificationVersion;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+
+/**
+ * Configure Maven for the desired version of Java.
+ *
+ * @author Basil Crow
+ */
+@Mojo(name = "initialize", defaultPhase = LifecyclePhase.INITIALIZE)
+public class InitializeMojo extends AbstractJenkinsMojo {
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        setCompilerProperties();
+    }
+
+    private void setCompilerProperties() throws MojoExecutionException {
+        if (!project.getProperties().containsKey("maven.compiler.source")
+                && !project.getProperties().containsKey("maven.compiler.release")) {
```

Comment:
This presumes that the POM is configuring these mojo parameters via properties, which is not necessary, and in fact discouraged—best configured via `<configuration>` in `<plugin>`.

More robust to look up the effective configuration of the mojo. Do not recall the idiom for that offhand.

Reply:
> More robust to look up the effective configuration of the mojo. Do not recall the idiom for that offhand.

Declined. If you want to understand the reason why then just try writing the code and you will quickly find out.